### PR TITLE
Kanvi/Add capability to get serialized graphs and other debug graphs

### PR DESCRIPTION
**Pull request by kanvi-nervana**
Add capability to get serialized graphs and other debug graphs while using unit test framework for debugging.

### CODE REVIEW

File: test/opexecuter.cpp [hunk 1]
```diff
@@ -312,6 +317,18 @@ void OpExecuter::ExecuteOnNGraph() {
   // ng function should get same number of outputs
   ASSERT_EQ(expected_output_datatypes_.size(), ng_function->get_output_size());
 
+  // For debug
+  // Serialize to nGraph if needed
+  if (std::getenv("NGRAPH_ENABLE_SERIALIZE") != nullptr) {
+    std::string file_name = "unit_test" + test_op_type_ + ".json";
+    NGRAPH_VLOG(0) << "Serializing graph to: " << file_name;
```
Comment (sayantan-nervana) [inline test/opexecuter.cpp]: currently the file name string is:
```unit_testAdd.json```
we could add an underscore and make it:
```unit_test_Add.json```

File: test/opexecuter.cpp [hunk 2]
```diff
@@ -312,6 +317,18 @@ void OpExecuter::ExecuteOnNGraph() {
   // ng function should get same number of outputs
   ASSERT_EQ(expected_output_datatypes_.size(), ng_function->get_output_size());
 
+  // For debug
+  // Serialize to nGraph if needed
+  if (std::getenv("NGRAPH_ENABLE_SERIALIZE") != nullptr) {
+    std::string file_name = "unit_test_" + test_op_type_ + ".json";
+    NGRAPH_VLOG(0) << "Serializing graph to: " << file_name;
+    std::string js = ngraph::serialize(ng_function, 4);
+    {
```
Comment (avijit-nervana) [inline test/opexecuter.cpp]: Why do you need this extra scope? Also - you may consider a try catch here i.e., to catch potential exceptions thrown by the ofstream <<. It's a good practice to catch exceptions especially during I/O.

Comment (sayantan-nervana) [reply test/opexecuter.cpp]: In which case, lets make similar changes here too:
https://github.com/NervanaSystems/ngraph-tf/blob/55f9f7ee133a94e05f41fedc4518b81a4c50563f/src/ngraph_encapsulate_op.cc#L277

Comment (avijit-nervana) [reply test/opexecuter.cpp]: Agree with @sayantan-nervana - please add this change in this PR.

File: src/ngraph_encapsulate_op.cc [hunk 3]
```diff
@@ -275,9 +275,14 @@ class NGraphEncapsulateOp : public OpKernel {
...
         std::string js = ngraph::serialize(ng_function, 4);
-        {
-          std::ofstream f(file_name);
+        std::ofstream f;
+        f.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+        try {
+          f.open(file_name);
           f << js;
+          f.close();
+        } catch (std::ofstream::failure e) {
```
Comment (avijit-nervana) [inline src/ngraph_encapsulate_op.cc]: 1. May consider adding `&` 
2. Add e.what() in the error message display.

File: test/opexecuter.cpp [hunk 4]
```diff
@@ -312,6 +317,23 @@ void OpExecuter::ExecuteOnNGraph() {
...
+    std::string file_name = "unit_test_" + test_op_type_ + ".json";
+    NGRAPH_VLOG(0) << "Serializing graph to: " << file_name << endl;
+    std::string js = ngraph::serialize(ng_function, 4);
+    std::ofstream f;
+    f.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+    try {
+      f.open(file_name);
+      f << js;
+      f.close();
+    } catch (std::ofstream::failure e) {
```
Comment (avijit-nervana) [inline test/opexecuter.cpp]: Same as above.

**Pull request merged**